sylt/src/formatter.rs

```diff
@@ -0,0 +1,504 @@
+        ExpressionKind::IfShort { condition, fail, lhs: _ } => {
+            write!(dest, "if ")?;
+            write_expression(dest, indent, condition)?;
+            write!(dest, " else ")?;
+            write_expression(dest, indent, fail)?;
+        }
+        ExpressionKind::Function {
+            name: _,
+            params,
+            ret,
+            body,
+        } => {
+            write!(dest, "fn")?;
+            if !params.is_empty() {
+                write!(dest, " ")?;
+            }
+            write_parameters(dest, indent, params)?;
+            if matches!(ret.kind, TypeKind::Resolved(RuntimeType::Void)) {
+                write!(dest, " ")?;
+            } else {
+                write!(dest, " -> ")?;
+                write_type(dest, indent, ret)?;
+                write!(dest, " ")?;
+            }
+            write_statement(dest, indent, body)?;
+        }
+        ExpressionKind::Instance { blob, fields } => {
+            write_assignable(dest, indent, blob)?;
+            write!(dest, " {{\n")?;
+            write_blob_instance_fields(dest, indent + 1, fields)?;
+            write_indents(dest, indent)?;
+            write!(dest, "}}")?;
+        }
+        ExpressionKind::Tuple(exprs) => {
+            write!(dest, "(")?;
+            if exprs.is_empty() {
+                write!(dest, ",")?;
+            } else {
+                write_comma_separated!(dest, indent, write_expression, exprs);
+            }
+            write!(dest, ")")?;
+        }
+        ExpressionKind::List(exprs) => {
+            write!(dest, "[")?;
+            write_comma_separated!(dest, indent, write_expression, exprs);
+            write!(dest, "]")?;
+        }
+        ExpressionKind::Set(exprs) => {
+            write!(dest, "{{")?;
+            write_comma_separated!(dest, indent, write_expression, exprs);
+            write!(dest, "}}")?;
+        }
+        ExpressionKind::Dict(exprs) => {
+            write!(dest, "{{")?;
+            let mut first = true;
+            let mut exprs = exprs.iter();
+            while let Some(expr) = exprs.next() {
+                if !first {
+                    write!(dest, ", ")?;
+                }
+                first = false;
+                write_expression(dest, indent, expr)?;
+                write!(dest, ": ")?;
+                write_expression(dest, indent, exprs.next().unwrap())?;
+            }
+            write!(dest, "}}")?;
+        }
+        ExpressionKind::Float(f) => write!(dest, "{}", f)?,
+        ExpressionKind::Int(i) => write!(dest, "{}", i)?,
+        ExpressionKind::Str(s) => write!(dest, "\"{}\"", s)?,
+        ExpressionKind::Bool(b) => write!(dest, "{}", if *b { "true" } else { "false" })?,
+        ExpressionKind::Nil => write!(dest, "nil")?,
+    }
+
+    Ok(())
+}
+
+fn write_statement<W: Write>(dest: &mut W, indent: u32, statement: &Statement) -> fmt::Result {
+    for comment in &statement.comments {
+        write!(dest, "// {}\n", comment)?;
```

Comment:
Strip the leading and trailing whitespace?

Reply:
Leading whitespace is stripped by the tokenizer but I'll add trim_end as well.